samples/extensions/dynamic_rendering_local_read/dynamic_rendering_local_read.cpp

```diff
@@ -0,0 +1,1052 @@
+/* Copyright (c) 2024, Sascha Willems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "dynamic_rendering_local_read.h"
+#include "scene_graph/components/mesh.h"
+#include "scene_graph/components/pbr_material.h"
+#include "scene_graph/components/sub_mesh.h"
+
+DynamicRenderingLocalRead::DynamicRenderingLocalRead()
+{
+	title = "Dynamic Rendering local read";
+
+	camera.type = vkb::CameraType::FirstPerson;
+	camera.set_position({0.0f, 0.5f, 0.0f});
+	camera.set_position({-3.2f, 1.0f, 5.9f});
+	camera.set_rotation({0.5f, 210.05f, 0.0f});
+	camera.set_perspective(60.f, static_cast<float>(width) / static_cast<float>(height), 256.f, 0.1f);
+
+#if defined(USE_DYNAMIC_RENDERING)
+	set_api_version(VK_API_VERSION_1_2);
+	add_instance_extension(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+	add_device_extension(VK_KHR_DYNAMIC_RENDERING_EXTENSION_NAME);
+	add_device_extension(VK_KHR_DYNAMIC_RENDERING_LOCAL_READ_EXTENSION_NAME);
+	// To simplify barrier setup used for dynamic rendering, we use sync2
+	add_device_extension(VK_KHR_SYNCHRONIZATION_2_EXTENSION_NAME);
+
+	// Dynamic rendering doesn't use render passes
+	// To make sure that framework related classes like the user interface are aware of this, we explicitly st the base class' renderpass to a null handle
+	render_pass = VK_NULL_HANDLE;
+
+	LOGI("Using dynamic rendering with local read");
+#endif
+}
+
+DynamicRenderingLocalRead::~DynamicRenderingLocalRead()
+{
+	if (device)
+	{
+		for (Pass pass : {scene_opaque_pass, scene_transparent_pass, composition_pass})
+		{
+			vkDestroyPipeline(get_device().get_handle(), pass.pipeline, nullptr);
+			vkDestroyPipelineLayout(get_device().get_handle(), pass.pipeline_layout, nullptr);
+			vkDestroyDescriptorSetLayout(get_device().get_handle(), pass.descriptor_set_layout, nullptr);
+		}
+		for (FrameBufferAttachment attachment : {attachments.albedo, attachments.normal, attachments.positionDepth})
```

Comment:
Same as above. But is it really a problem? It's the destructor anyway, so the copy shouldn't hurt.

Reply:
Indeed, same as above.
And it might be a bad habit of mine, but I don't like code that unnecessarily wastes cycles. Even if it's "just" the destructor. Especially if it's relatively easy to resolve. And besides that, that `destroy_attachment` is not only called in the destructor.